### DISCUSSION TITLE
may PSV rev be a possible option？

### DISCUSSION THREAD

**Discussion by maskmark** [xfangfang/wiliwili#87] [General]
and ETA wen ：）shuanQ buddy

**Comment by xfangfang** [comment]
@maskmark 这英语要不是懂点汉语还真的看不太懂😀

PSV从内存和性能上看移植的可能性不高。

从技术上讲，是完全可以做到的，PSV有 ffmpeg，支持OpenGL ES2，有移植的GLFW，甚至 wiliwili 用到的底层UI库也可以几乎正常地运行在PSV上。

但是存在一些问题：

- 视频方面
  如果用ffmpeg的软解可能会很卡。硬解的支持需要写很多播放器专属的代码（能播放的上限应该在720P-1080P）
- 内存方面
  512MB内存 128MB显存，对 wiliwili 来说可能会需要很多本地的缓存来临时储存屏幕上的图片（这一部分是需要重新实现的）
- 性能方面
  我使用树莓派3B运行 wiliwili 会很卡，我怀疑 PSV 也不能够满分辨率运行起来这个应用

从上面的信息可以知道，移植的可能性是有的，需要花很多时间（就像在做switch的客户端一样，破解机自制软件会有很多坑且没有文档可查）考虑到移植的收益，所以我不认为会有人做这样的移植。

**Reply by maskmark** [reply]
Right. quite a clear explanation.so let it be a vaporware in my dream : ) THX anyway.
